test/e2e/support/utils.js

```diff
@@ -475,3 +476,127 @@ export function observeFocusLoss() {
 		} );
 	} );
 }
+
+/**
+ * Creates a function to determine if a request is embedding a certain URL.
+ *
+ * @param {string} url The URL to check against a request.
+ * @return {function} Function that determines if a request is for the embed API, embedding a specific URL.
+ */
+export function isEmbedding( url ) {
+	return ( request ) => matchURL( 'oembed%2F1.0%2Fproxy' )( request ) && parameterEquals( 'url', url )( request );
+}
+
+/**
+ * Respond to a request with a JSON response.
+ *
+ * @param {string} mockResponse The mock object to wrap in a JSON response.
+ * @return {Promise} Promise that responds to a request with the mock JSON response.
+ */
+export function JSONResponse( mockResponse ) {
+	return async ( request ) => request.respond( getJSONResponse( mockResponse ) );
+}
+
+/**
+ * Creates a function to determine if a request is calling a URL with the substring present.
+ *
+ * @param {string} substring The substring to check for.
+ * @return {function} Function that determines if a request's URL contains substring.
+ */
+export function matchURL( substring ) {
+	return ( request ) => -1 !== request.url().indexOf( substring );
+}
+
+/**
+ * Creates a function to determine if a request has a parameter with a certain value.
+ *
+ * @param {string} parameterName The query parameter to check.
+ * @param {string} value The value to check for.
+ * @return {function} Function that determines if a request's query parameter is the specified value.
+ */
+export function parameterEquals( parameterName, value ) {
+	return ( request ) => {
+		const url = request.url();
+		return value === decodeURIComponent( new RegExp( '.*' + parameterName + '=([^&]+).*' ).exec( url )[ 1 ] );
+	};
+}
+
+/**
+ * Get a JSON response for the passed in object, for use with `request.respond`.
+ *
+ * @param {Object} obj Object to seralise for response.
+ * @return {Object} Response for use with `request.respond`.
+ */
+export function getJSONResponse( obj ) {
+	return {
+		content: 'application/json',
+		body: JSON.stringify( obj ),
+	};
+}
+
+/**
+ * Mocks a request with the supplied mock object, or allows it to run with an optional transform, based on the
+ * deserialised JSON response for the request.
+ *
+ * @param {function} mockCheck function that returns true if the request should be mocked.
+ * @param {Object} mock A mock object to wrap in a JSON response, if the request should be mocked.
```

Comment:
I think we usually line these up, but _personally_ I don't really mind if you don't. It'd be nice if we had a lint rule for it, but it's a rare instance where I'm not personally wildly-picky about code styles 😆